integrationtests/as-lucene-directory/src/test/resources/user-provided-infinispan-persistence.xml

```diff
@@ -10,8 +10,12 @@
         <jmx duplicate-domains="true"/>
 
         <replicated-cache name="LuceneIndexesMetadata" mode="SYNC" remote-timeout="25000">
+            <transaction mode="NONE" />
             <persistence passivation="false">
-                <string-keyed-jdbc-store preload="true" key-to-string-mapper="org.infinispan.lucene.LuceneKey2StringMapper" shared="true">
+                <!-- N.B. "shared" is false which is unusual for a JDBC CacheStore,
+                 but for testing purposes we use an in-memory H2 so each WildFly node
+                 will have an independent CacheStore instance -->
+                <string-keyed-jdbc-store preload="true" key-to-string-mapper="org.infinispan.lucene.LuceneKey2StringMapper" shared="false">
```

Comment:
It used to be like that, with an H2 configuration to store the database in a file on ~.

But then we had sporadic failures caused by H2 non correctly wiping the database.. 

I think this is also totally valid: we're testing both a non-shared  cacheStore (people might use one?) and we're testing that we're able to configure a JDBC CacheStore, which was the main issue I wanted to resolve as that was broken.

I agree it's unlikely for someone to *combine* the JDBC CacheStore with a non-shared-store, but it's just a test which should pass. The comment addresses that.